tidy3d/components/medium.py

```diff
@@ -306,76 +336,283 @@ def from_nk(cls, n: float, k: float, freq: float, **kwargs):
+        at ``frequency`` interpolated at the supplied coordinates.
+
+        Parameters
+        ----------
+        frequency : float
+            Frequency to evaluate permittivity at (Hz).
+        coords : :class:`.Coords`
+            The grid point coordinates over which interpolation is performed.
+
+        Returns
+        -------
+        Tuple[Numpy, Numpy, Numpy]
+            The complex-valued permittivity tensor at ``frequency`` interpolated
+            at the supplied coordinate.
+        """
+
+        eps_freq = self.eps_dataset_freq(frequency)
+        interp_shape = [len(coord_comp) for coord_comp in coords.to_list]
+        eps_list = [
+            np.array(
+                self._interp(eps_freq.field_components[comp], coords, self.interp_method)
+            ).reshape(interp_shape)
+            for comp in ["eps_xx", "eps_yy", "eps_zz"]
+        ]
+        return tuple(eps_list)
 
     @ensure_freq_in_range
     def eps_diagonal(self, frequency: float) -> Tuple[complex, complex, complex]:
-        """Main diagonal of the complex-valued permittivity tensor as a function of frequency."""
+        """Main diagonal of the complex-valued permittivity tensor
+        at ``frequency``. Spatially, we take max{|eps|}, so that autoMesh generation
+        works appropriately.
+        """
+        eps_freq = self.eps_dataset_freq(frequency)
+        eps_np_list = [
+            np.array(eps_freq.field_components[comp]).ravel()
+            for comp in ["eps_xx", "eps_yy", "eps_zz"]
+        ]
+        eps_list = [eps_comp[np.argmax(np.abs(eps_comp))] for eps_comp in eps_np_list]
+        return tuple(eps_list)
 
-        eps_xx = self.xx.eps_model(frequency)
-        eps_yy = self.yy.eps_model(frequency)
-        eps_zz = self.zz.eps_model(frequency)
-        return (eps_xx, eps_yy, eps_zz)
+    @ensure_freq_in_range
+    def eps_model(self, frequency: float) -> complex:
+        """Spatial and poloarizaiton average of complex-valued permittivity
+        as a function of frequency.
+        """
+        eps_freq = self.eps_dataset_freq(frequency)
+        eps_arrays = [eps_freq.field_components[comp] for comp in ["eps_xx", "eps_yy", "eps_zz"]]
+        eps_array_avgs = [np.mean(eps_array) for eps_array in eps_arrays]
+        return np.mean(eps_array_avgs)
 
-    @add_ax_if_none
-    def plot(self, freqs: float, ax: Ax = None) -> Ax:
-        """Plot n, k of a :class:`Medium` as a function of frequency."""
+    @classmethod
+    def from_eps_raw(
+        cls, eps: ScalarFieldDataArray, interp_method: InterpMethod = "nearest"
+    ) -> CustomMedium:
+        """Construct a :class:`CustomMedium` from datasets containing raw permittivity values.
 
-        freqs = np.array(freqs)
-        freqs_thz = freqs / 1e12
+        Parameters
+        ----------
+        eps : :class:`.ScalarFieldDataArray`
+            Dataset containing complex-valued permittivity as a function of space.
+        interp_method : InterpMethod, optional
```

Comment:
might want to provide the raw literal values accepted here, or :class: link to InterpMethod if that works.

Reply:
As for the raw literal values, is it like this:
`interp_method : ["nearest", "linear"], optional`